### PR TITLE
Split styles into modules

### LINKED ISSUE #1: Split styles into modules
[Current styles](https://github.com/voorhoede/demo-viewer/blob/b7ff26/src/index.css) are directly copied from [voorhoede/front-end-guide](https://github.com/voorhoede/front-end-guide). These should be split into modules (`info-panel`, `module-selector`, etc). Use a pre-processor (eg. less or postcss).

### PR DESCRIPTION
**Pull request by reinoute**
Resolves #1 and #7

### CODE REVIEW

File: src/modules/app/app.css [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+}
+
+.app-header {
+    padding: 0 0.5em;
+    background-color: var(--color-grey);
+    border-bottom: 1px solid var(--color-grey-dark);
+    box-shadow: 0 0 5px rgba(0, 0, 0, 0.2);
+}
+
+.app-header > *:not(:last-child) {
```
Comment (jbmoelker) [inline src/modules/app/app.css]: `.app-header > * + * { margin-left: ... }` does the same thing but is more expressive I think. As what it says is: "add some space before the next element". That's what a designer would say. Not "add some space after each element except for the last element". Also `* + *` has better browser support, but doesn't matter for IE9+ scope.

File: src/modules/core/core.css [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+    line-height: inherit;
+    text-decoration: none;
+    text-align: center;
+    font-weight: bold;
+    color: var(--color-blue-dark);
+    vertical-align: top;
+    box-shadow: 0 0 3px rgba(0, 0, 0, 0.2);
+}
+
+.button:hover {
```
Comment (jbmoelker) [inline src/modules/core/core.css:78]: also `.button:focus`?

File: src/modules/core/core.css [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+    background-color: var(--color-grey-dark);
+    cursor: pointer;
+}
+
+.button.is-active {
+    background-color: var(--color-blue-dark);
+    color: var(--color-yellow);
+}
+
+.button.is-active:hover {
```
Comment (jbmoelker) [inline src/modules/core/core.css:89]: also `.button.is-active:focus`

File: src/modules/icon/icon.css [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+
+.icon-component:before {
+    content: '\e628';
+}
+
+.icon-align-top:before {
+    content: '\e7a7';
+}
+
+.ie8 .icon-align-top {
```
Comment (jbmoelker) [inline src/modules/icon/icon.css]: All IE8 styles can be removed. The Riot version used (2.3+) no longer supports IE8. When testing in IE8 the developer will just have to open the individual demo pages without the viewer.

File: src/modules/info-panel/info-panel.css [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+.info-panel-header {
+    padding: 0.25em;
+    line-height: 2;
+}
+
+.info-panel-navigation {
+    display: inline-block;
+}
+
+.info-panel-navigation > *:not(:last-child) {
```
Comment (jbmoelker) [inline src/modules/info-panel/info-panel.css]: Again `> * + *` or maybe `> li + li` in this case?

File: src/modules/info-panel/info-panel.css [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+.info-panel-control {
+    float: right;
+}
+
+.info-panel-control .button {
+    width: 42px;
+    padding: 0;
+}
+
+.info-panel-control > *:not(:last-child) {
```
Comment (jbmoelker) [inline src/modules/info-panel/info-panel.css]: Again `> * + *` in this case?

File: src/modules/info-panel/info-panel.tag.html [hunk 7]
```diff
@@ -1,27 +1,33 @@
-<info-panel class="viewer-info" style="height:60%;">
+<info-panel>
```
Comment (jbmoelker) [inline src/modules/info-panel/info-panel.tag.html]: also add aria-controls on the "info button" in the top toolbar and link it to this panel? Will also require `aria-expanded`.

File: src/modules/module-selector/module-selector.css [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+@import '../icon/icon.css';
+
+.module-selector-select {
+    height: 42px; /* line-height has no effect on replaced elements */
+    margin-right: var(--default-margin);
+    background-color: var(--color-grey-light);
+    border: 1px solid transparent;
+}
+
+.module-selector-select:hover {
```
Comment (jbmoelker) [inline src/modules/module-selector/module-selector.css:15]: Add `:focus`?

File: src/modules/size-controls/size-controls.css [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+    border: 1px solid transparent;
+}
+
+.size-controls-input:hover {
+    transition: all 0.2s ease-in;
+    border: 1px solid var(--color-blue-dark);
+    background-color: var(--color-grey);
+}
+
+@media screen and (min-width: 525px) {
```
Comment (jbmoelker) [inline src/modules/size-controls/size-controls.css]: I think showing or hiding the "size-controls" makes more sense to define on the level of the app. The app knows when they will fit or not in relation to other content of the top toolbar.

File: examples/bootstrap-v3/index-json.html [hunk 10]
```diff
@@ -5,6 +5,11 @@
     <meta name="viewport" content="width=device-width">
     <title>Demo viewer example: Bootstrap v3</title>
 </head>
+<style type="text/css">
+body {
```
Comment (jbmoelker) [inline examples/bootstrap-v3/index-json.html]: move to `src/index.css`

File: src/modules/app/app.css (hunk 1 above)
Comment (reinoute) [inline src/modules/app/app.css]: fixed

File: src/modules/core/core.css (hunk 2 above)
Comment (reinoute) [inline src/modules/core/core.css:78]: see line 44.

File: src/modules/core/core.css (hunk 3 above)
Comment (reinoute) [inline src/modules/core/core.css:89]: see line 44. I use outline for focus, so I don't need to redefine the focus styles every time.

File: src/modules/icon/icon.css (hunk 4 above)
Comment (reinoute) [inline src/modules/icon/icon.css]: done

File: src/modules/info-panel/info-panel.css (hunk 5 above)
Comment (reinoute) [inline src/modules/info-panel/info-panel.css]: done

File: src/modules/info-panel/info-panel.css (hunk 6 above)
Comment (reinoute) [inline src/modules/info-panel/info-panel.css]: done

File: src/modules/info-panel/info-panel.tag.html (hunk 7 above)
Comment (reinoute) [inline src/modules/info-panel/info-panel.tag.html]: Ok if I fix this in #6? I just started on that.

File: src/modules/module-selector/module-selector.css (hunk 8 above)
Comment (reinoute) [inline src/modules/module-selector/module-selector.css:15]: already covered by core styles (line 44)

File: examples/bootstrap-v3/index-json.html (hunk 10 above)
Comment (reinoute) [inline examples/bootstrap-v3/index-json.html]: done

File: src/modules/size-controls/size-controls.css (hunk 9 above)
Comment (reinoute) [inline src/modules/size-controls/size-controls.css]: True!

**Comment by jbmoelker** [comment]
Code looks good. 

Add CSS sourcemaps? Now its hard to debug as everything is on line 1 of dist file (`demo-viewer.css:1`).

Fine tune styling of controls:

Size controls are now:
screen shot 2016-05-09 at 17 33 59

They could do with a bit less spacing, the "full" text overflows and could be a bit smaller, the hover of the input is a bit awkward to me, and I would put the "px" on the same color background as the input, and note the font size and weight of the input and "px". So basically update to look like they did before:

screen shot 2016-05-09 at 17 33 28

Same holds for the module selector and options:

screen shot 2016-05-09 at 17 34 07

Make select look square with custom caret, spacing of buttons like in size control. So:

screen shot 2016-05-09 at 17 33 37

(yes, in the last screenshot the buttons shouldn't overlap :) )

The size controls disappear quite early. I think they could be visible on smaller screens until they meet the module selector and info buttons.

**Comment by reinoute** [comment]
Done, see latest changes.

**Comment by jbmoelker** [comment]
Looks really good! Better than it currently does in the front-end-guide I think.

Just have one very little thing. I'd add `.size-controls-input { text-align: right; }`.

https://github.com/voorhoede/demo-viewer/pull/14/files#diff-f9212de3d84c5d258b8aff808972eab7R35

**Comment by reinoute** [comment]
Done.

**Pull request merged**